diag_manager/fms_diag_yaml.F90

```diff
@@ -1667,13 +1667,14 @@ subroutine fms_diag_yaml_out()
   call fms_f2c_string( keys(1)%key2, 'base_date')
   basedate_loc = diag_yaml%get_basedate()
   tmpstr1 = ''; tmpstr2 = ''
-  tmpstr1 = string(basedate_loc(1))
+  tmpstr1 = '[ '//string(basedate_loc(1))
```

Comment:
Since the new bracket format is part of the yaml specification the spacing shouldn't matter, I did a little testing just to make sure and trailing/leading spaces will get ignored. This string specifically will be broken down into elements before they are passed to the libyaml library for writing, so the library handles adding the brackets, commas and any spaces to the output.